src/sources/syslog.rs

```diff
@@ -242,101 +233,38 @@ pub fn udp(
     })
 }
 
-/// Function used to resolve the year for syslog messages that don't include the year.
-/// If the current month is January, and the syslog message is for December, it will take the previous year.
-/// Otherwise, take the current year.
-fn resolve_year((month, _date, _hour, _min, _sec): IncompleteDate) -> i32 {
-    let now = Utc::now();
-    if now.month() == 1 && month == 12 {
-        now.year() - 1
-    } else {
-        now.year()
-    }
-}
+fn event_from_bytes(host_key: &str, default_host: Option<Bytes>, bytes: Bytes) -> Option<Event> {
+    let byte_size = bytes.len();
+    let parser = SyslogParser;
+    let events = parser.parse(bytes).ok()?;
+    let mut event = events[0].clone();
```

Comment:
This doesn't seem like a great pattern. Do we want to assert that there will only every be a single event parsed? It could be clearer to rearrange things a bit and `map` the rest of this function over the output of the parser.

Reply:
In this case specifically, the assertion makes sense:

https://github.com/timberio/vector/blob/53e90c6a989e37cdd0982b4372b9bdf26da7b89d/src/codecs/parsers/syslog.rs#L40

Since `syslog_loose::parse_message` only ever returns one message.

However, there's some improvement I can see here: We can check the length and use `Vec::remove` to avoid the clone, and return `None` otherwise